nucypher/blockchain/eth/deployers.py

```diff
@@ -654,22 +674,39 @@ def deploy(self,
             progress.update(1)
 
         # 4 - Initialize the StakingEscrow contract
-        init_function = the_escrow_contract.functions.initialize(self.economics.erc20_reward_supply)
-
+        init_function = self._contract.functions.initialize(self.economics.erc20_reward_supply)
         init_receipt = self.blockchain.send_transaction(contract_function=init_function,
                                                         sender_address=self.deployer_address,
                                                         payload=origin_args)
         if progress:
             progress.update(1)
 
-        # Gather the transaction receipts
-        ordered_receipts = (deploy_receipt, dispatcher_deploy_receipt, approve_reward_receipt, init_receipt)
-        deployment_receipts = dict(zip(self.deployment_steps, ordered_receipts))
+        activation_receipts = dict(zip(self.activation_steps, (approve_reward_receipt, init_receipt)))
+        return activation_receipts
 
-        # Set the contract and transaction receipts #
-        self._contract = the_escrow_contract
-        self.deployment_receipts = deployment_receipts
-        return deployment_receipts
+    @property
+    def ready_to_activate(self) -> bool:
```

Comment:
I guess it's duplicated in the next method `is_active`, maybe remove this and keep other method?

Reply:
Not really, this method deals with `current_supply_1 == 0`, while the next one is `current_supply_1 != 0`.

Reply:
I see, this confuses, especially `ready_to_activate` and 
 when `current_supply_1 != 0` which means contract activated and never will be ready to activate again